src/main/scala/ai/privado/entrypoint/ScanProcessor.scala

```diff
@@ -260,19 +268,11 @@ object ScanProcessor extends CommandProcessor {
     val sinkSkipList = externalConfigAndRules.sinkSkipList ++ internalConfigAndRules.sinkSkipList
     val systemConfig = externalConfigAndRules.systemConfig ++ internalConfigAndRules.systemConfig
     val auditConfig  = externalConfigAndRules.auditConfig ++ internalConfigAndRules.auditConfig
-    val combinedSources = sources
-      .groupBy(_.id)
-      .view
-      .map { case (_, sourceGroup) =>
-        val combinedPattern = sourceGroup.map(_.patterns.mkString("|")).mkString("(", "|", ")")
-        sourceGroup.head.copy(patterns = List(combinedPattern))
-      }
-      .toList
     val mergedRules =
       ConfigAndRules(
-        sources = combinedSources,
-        sinks = sinks.distinctBy(_.id),
-        collections = collections.distinctBy(_.id),
+        sources = mergePatterns(sources),
+        sinks = mergePatterns(sinks),
+        collections = mergePatterns(collections),
         policies = policies.distinctBy(_.id),
```

Comment:
@Saumya25mis Apply mergePatterns on `policies`, `exclusions` and `threats` as well